src/services/infra/DynamoDBClient.ts

```diff
@@ -0,0 +1,130 @@
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb"
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  GetCommand,
+  UpdateCommand,
+  DeleteCommand,
+  UpdateCommandInput,
+  PutCommandOutput,
+  UpdateCommandOutput,
+  DeleteCommandOutput,
+} from "@aws-sdk/lib-dynamodb"
+import autoBind from "auto-bind"
+
+import { MessageBody } from "@root/../microservices/site-launch/shared/types"
+import config from "@root/config/config"
+import logger from "@root/logger/logger"
+
+export interface UpdateParams {
+  TableName: string
+  Key: Record<string, string>
+  UpdateExpression: string
+  ExpressionAttributeNames: Record<string, string>
+  ExpressionAttributeValues: Record<string, string>
+}
+export default class DynamoDBDocClient {
+  private readonly dynamoDBDocClient: DynamoDBDocumentClient
+
+  constructor() {
+    const marshallOptions = {
+      // Whether to automatically convert empty strings, blobs, and sets to `null`.
+      convertEmptyValues: false,
+      // Whether to remove undefined values while marshalling.
+      removeUndefinedValues: true,
+      // Whether to convert typeof object to map attribute.
+      convertClassInstanceToMap: false,
+    }
+
+    const unmarshallOptions = {
+      // Whether to return numbers as a string instead of converting them to native JavaScript numbers.
+      wrapNumbers: false,
+    }
+
+    this.dynamoDBDocClient = DynamoDBDocumentClient.from(
+      new DynamoDBClient({ region: config.get("aws.amplify.region") }),
+      { marshallOptions, unmarshallOptions }
+    )
+
+    autoBind(this)
+  }
+
+  createItem = async (
+    tableName: string,
+    item: MessageBody
+  ): Promise<PutCommandOutput> => {
+    const params = {
+      TableName: tableName,
+      Item: {
+        PRIMARY_KEY: item.appId,
+        ...item,
+      },
+    }
+    try {
+      const response = await this.dynamoDBDocClient.send(new PutCommand(params))
+      return response
+    } catch (error) {
+      logger.error(`Error in createItem for the table ${tableName}: ${error}`)
+      throw error
+    }
+  }
```

Comment:
feels like the `try/catch` is placed at the wrong site. because we want to catch errors at their source (`client.send` here), maybe we could consider doing something similar to the following

 ```ts
  private withLogger = async <T>(
    p: Promise<T>,
    tableName: string
    type: "create" | "get" | "delete" | "update"
  ): Promise<T> => {
    try {
      return p
    } catch (e) {
      logger.error(`Error in ${type}Item for the table ${tableName}: ${e}`)
      throw e
    }
  }
```

seems to work fine and we can simplify our method to 
```ts
  createItem = async (
    tableName: string,
    item: MessageBody
  ): Promise<PutCommandOutput> => {
    const params = {...}

    return this.withLogger(
      this.dynamoDBDocClient.send(new PutCommand(params)),
      tableName
    )
  }
```

Reply:
Thanks for the clear suggestion! implemented [here](e76d7665)

Reply:
oh right opps good catch, done here : 761fc5a9b

Reply:
761fc5a9b26a5b15fd77f5608ae4a4949221f081